tests/test_engine_logic.py

```diff
@@ -253,12 +260,21 @@ def test_do_actions_with_injected_parameters(self):
 
             defined_actions = BaseActions()
             defined_actions.action1 = MagicMock()
+            defined_actions.action1.params = []
             defined_actions.action2 = MagicMock()
-            defined_actions.action2.params = {
-                'param1': fields.FIELD_TEXT,
-                'param2': fields.FIELD_NUMERIC
-            }
-
+            defined_actions.action2.params = [{
```

Comment:
Shouldn't this params be instances of `ActionParam`? `getattr` doesn't work for dictionaries.
```python
In [8]: d
Out[8]:
{'label': 'action2',
 'name': 'param2',
 'fieldType': 'field type',
 'defaultValue': None}

In [9]: getattr(d, 'fieldType')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-7d9db41d484f> in <module>
----> 1 getattr(d, 'fieldType')

AttributeError: 'dict' object has no attribute 'fieldType'
```

Reply:
Also, maybe you should keep both cases with both types of params definitions